clayui.com/plugins/gatsby-source-liferay/gatsby-node.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * SPDX-FileCopyrightText: © 2024 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const {Headers, fetch} = require('fetch-undici');
+
+const fetchLiferay = async (slug, siteId, username, host) => {
+	return await fetch(
+		new URL(
+			`/o/headless-delivery/v1.0/sites/${siteId}/site-pages${slug}`,
+			host
+		),
+		{
+			headers: new Headers({
+				Authorization: `Basic ${btoa(username)}`,
+				'User-Agent': 'Clay',
+				'content-type': 'application/json',
+			}),
+		}
+	)
+		.then(async (response) => {
+			const {status} = response;
+
+			const responseContentType = response.headers.get('content-type');
+
+			if (status === 204) {
+				return status;
+			} else if (
+				response.ok &&
+				responseContentType === 'application/json'
+			) {
+				return response.json();
+			} else {
+				const data = await response.text();
+
+				if (data.includes('NOT_FOUND')) {
```

Comment:
Actually I don't think we need this anymore, I added this when I was checking if a page exists which is strange that when a page is not found the `content-type` is a string instead of a json so I was doing this to decide whether to deserialize the JSON otherwise was HTML content. As we now only search for pages that exist, this check is no longer necessary.

I think this happened because the API did not add the header indicating that it was `application/json` content.